src/logging_custom_prefix_unittest.cc

```diff
@@ -1379,3 +1379,21 @@ TEST(UserDefinedClass, logging) {
   // We must be able to compile this.
   CHECK_EQ(u, u);
 }
+
+TEST(LogMsgTime, gmtoff) {
+  /*
+   * Unit test for GMT offset API
+   * TODO: To properly test this API, we need a platform independent way to set time-zone.
+   * */
+  google::LogMessage log_obj(__FILE__, __LINE__);
+
+  struct::tm time_struct;
+  time_t ts = static_cast<time_t>(-1);
+  int32_t usecs = -1;
+  log_obj.GetTimeInfo(time_struct, ts, usecs);
+
+  LogMessageTime time_obj(time_struct, ts, usecs);
```

Comment:
To be honest, splitting the API into `GetTimeInfo` followed by construction of `LogMessageTime` on the user side generates unnecessary boilerplate. Doesn't it make more sense to provide a `time` getter in `LogMessage` that returns a `LogMessageTime` instance and encapsulates both steps?

Reply:
Hi, 
`GetTimeInfo` is only used for unit testing. In the standard flow `LogMessageInfo` will instantiate `LogMessageTime`. 

I'm only worried about adding an extra data member to `LogMessage`. As the `LogMessageTime` data is only useful till we populate the `stream()`. 

Thanks,
Vijay

Reply:
That's inconsistent though creating a discrepancy between test logic and runtime behavior.

Reply:
Hi, 

I tried to add `LogMessageTime` as a data member in `LogMessage` ; it works for the regular cases but in case of FATAL(GLOG_FATAL) we aren't allowed to allocate with new. So having a static object to handle the fatal case wouldn't be cleaner approach (Construct the `LogMessageTime` with default values to members and later during `LogMessage::Init()` set the member values of `LogMessageTime`). 


Hence I decided to go with the alternate method you suggested; "Testing the gmtoff() method within the supplied custom prefix callback".  
Currently, in the file  `"src/logging_custom_prefix_unittest.cc"` we are passing `std::string` as callback data.  If I try to set a fresh value (Let's say the validity of gmtoffset) in this callback data,  the test `DeathNoAllocNewHook.logging` fails (`SegFault`) as the string assignment calls the new memory alloc which is not allowed in case of FATAL. 

Would it make sense to pass `integer` data instead of `std::string` for the callback data in the unittest? 

Thanks, 
Vijay

Reply:
Thanks for update. However, I'm not sure I understand why adding `LogMessageTime` as `LogMessage` member is not possible. Can you elaborate?

In fact, this is the preferred solution. Everything else is a compromise that will be eventually discarded.

Reply:
You mean you allocated `LogMessageTime` on heap? Why pointer to begin with?

Reply:
Yes. 
I used pointer because `LogMessageTime's` data members are const references and they hold time info. which is calculated in `LogMessage::Init()`. If I don't use pointer and just use object, I need to supply values to `LogMessageTime's` members in the `LogMessage's` constructor through initializer list but the values are not available yet.  Those will be calculated in `Init()` method called later by `LogMessage's` constructors

Reply:
You need to store the time as copy otherwise you'll end up with dangling references.

Reply:
ok